src/nomad_measurements/ppms/parser.py

```diff
@@ -0,0 +1,131 @@
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from time import perf_counter, sleep
+from typing import (
+    TYPE_CHECKING,
+)
+
+from nomad.datamodel import EntryArchive
+from nomad.datamodel.data import (
+    EntryData,
+)
+from nomad.datamodel.metainfo.annotations import (
+    ELNAnnotation,
+)
+from nomad.metainfo import Quantity
+from nomad.parsing import MatchingParser
+from nomad.search import search
```

Comment:
Can you try putting this import statement inside the function where you use search?

Reply:
Thank you, it indeed worked.

If you see in this branch, I moved the test file for XRD in a dedicated folder. I will need to change the paths in the test.

IS it okay for you to organize the test files in subfolders @ka-sarthak ?